crates/wiggle/generate/src/config.rs

```diff
@@ -61,14 +61,21 @@ impl Parse for ConfigField {
             input.parse::<Token![async]>()?;
             input.parse::<Token![:]>()?;
             Ok(ConfigField::Async(AsyncConf {
-                blocking: false,
+                block_with: None,
                 functions: input.parse()?,
             }))
         } else if lookahead.peek(kw::block_on) {
             input.parse::<kw::block_on>()?;
+            let block_with = if input.peek(syn::token::Bracket) {
+                let content;
+                let _ = bracketed!(content in input);
+                content.parse()?
+            } else {
+                quote::quote!(wiggle::run_in_dummy_executor)
+            };
```

Comment:
thanks this is a really good idea. we should do the exact same in wasmtime-wit-bindgen.